libs/marinade-client-rs/src/transactions/signature_builder.rs

```diff
@@ -47,9 +62,16 @@ impl SignatureBuilder {
             if let Some(keypair) = self.signers.get(&key) {
                 transaction.signatures[pos] = keypair.try_sign_message(&message)?;
             } else {
-                error!("sign_transaction: not enough signers, expected key: {}, available keys in builder: {:?}",
-                    key, self.signers.keys().collect::<Vec<&Pubkey>>());
-                return Err(SignerError::NotEnoughSigners);
+                let error_msg = format!(
+                    "sign_transaction: not enough signers, expected key: {}, available keys in builder: {:?}",
+                    key, self.signers.keys().collect::<Vec<&Pubkey>>()
+                );
+                if self.is_check_signers {
+                    error!("{}", error_msg);
+                    return Err(SignerError::NotEnoughSigners);
+                } else {
+                    debug!("{}", error_msg);
```

Comment:
(non-blocking)
Is this debug log truly needed? If not, it would feel more natural to me to put the `if` with `is_check_signers` above the attempt to sign, rather than deciding after an error occurs (not a big deal though).

Side note (unrelated to this PR): Returning `NotEnoughSigners` upon any error seems suspicious, but I haven't explored further.

Reply:
ad debug) Thanks. To my understanding and usage of the code the `debug` message is usable as one can see (even when it's not considered as error flow).
I belive it's probably a bit different way I'm used to look through the processing and this kind of stuff I expect to get on printing to debug (or maybe to trace).
By me I would leave it as it is.

ad NotEnoughSigners) what do you mean suspicious? It could be just I'm not getting it fully. The processing has got "inspired" from the `try_sign" of the `Signature` https://docs.rs/solana-sdk/latest/solana_sdk/signer/trait.Signer.html#method.sign_message
When the transaction cannot be signed then the error is thrown.